### ISSUE TITLE
Health Check fails on plugin Reconfigure and downloaded bundles are Not Modified

### ISSUE THREAD

**Issue by asadk12** [open-policy-agent/opa#6312]
## Short description 
We are running a bundle api server distributing to OPA deployed as a sidecar across kubernetes. We run across a niche use case where the server changes configuration which when pulled by the agent makes it unhealthy **If the bundles are Not Modified** because on the health check we are using `/health?bundles=true` as mentioned in the [documentation](https://www.openpolicyagent.org/docs/latest/deployments/#readiness-and-liveness-probes)

### Example
- discovery bundle response before change
```json
{
    "result": {
        "bundles": {
            "my_policy": {
                "resource": "v1/bundles/my_policy",
                "service": "main"
            },
        },
        "default_decision": "authz"
    }
}
```
- discovery bundle response after change
```json
{
    "result": {
        "bundles": {
            "my_policy": {
                "polling":{
                    "min_polling_duration": 30,
                    "max_polling_duration": 60
                },
                "resource": "v1/bundles/my_policy",
                "service": "main"
            },
        },
        "default_decision": "authz"
    }
}
```

* OPA version: tested from 0.49.0 - HEAD
* In the example above the discovery provides an update for the bundle config
* Error Message: "{\"error\":\"one or more bundles are not activated\"} 
* health check HTTP response 500 for `/health?bundles=true`

## Steps To Reproduce

- Please see #6311 adding a test to ensure state and proposed state update

1. When discovery update is processed it trigger `plugins.Reconfigure(...)` which explicitly marks the state as NotReady, the expectation is for the downloaders to kick in and reconcile the state to StateOK
2. There is a change in bundle config such as polling or resource but when the downloader pulls the bundle it gets a 304 because the bundle itself is **Not Modified**
3. The manager fails to update the state to `plugins.StateOK` due to which the health check starts returning 500
4. [This part of the code](https://github.com/open-policy-agent/opa/blob/main/plugins/bundle/plugin.go#L540) should be updating  the plugin state in this specific use case

## Expected behavior
The bundle manager should be in StateOK.

**Comment by ashutosh-narkar** [comment]
[PR](https://github.com/open-policy-agent/opa/pull/6311) to fix this bug.

**Comment by asadk12** [comment]
fixed by #6311
